src/pwm.rs

```diff
@@ -44,7 +44,8 @@ where
     pub fn new(timer: I, frequency: impl Into<Hertz>, rcc: &mut Rcc) -> Self {
         let frequency = frequency.into();
 
-        timer.enable(rcc);
+        I::enable(rcc);
+        I::reset(rcc);
```

Comment:
Note: At first glance, this looks like a reset was added, but the reset was already present in the previous `enable` method. Looks correct to me.